File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.py
class Solution:
    def pivotInteger(self, n: int) -> int:
        precompute = [0] * 1001

        if precompute[1] == 0:
            for i in range(1, 1001):
                sum_val = i * (i + 1) // 2

                j = 1
                while j * j < sum_val:
                    j += 1

                precompute[i] = j if j * j == sum_val else -1

        return precompute[n]

File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.cpp
class Solution {
public:
    int pivotInteger(int n) {
        int leftPointer = 1;
        int rightPointer = n;
        int sumLeft = leftPointer;
        int sumRight = rightPointer;

        if(n==leftPointer)return n;
        // Iterate until the pointers meet
        while (leftPointer < rightPointer) {
            // Adjust sums and pointers based on comparison
            if (sumLeft < sumRight) {
                sumLeft += ++leftPointer;
            } else {
                sumRight += --rightPointer;
            }

            // Check for pivot condition
            if (sumLeft == sumRight && n !=leftPointer && leftPointer + 1 == rightPointer - 1) {
                return leftPointer + 1;
            }
        }

        return -1; // Return -1 if no pivot is found
    }
};